### ISSUE TITLE
Cannot reproduce the results from the project page

### ISSUE THREAD

**Issue by hollance** [nikhilsinghmus/room2reverb#32]
Hi, I was trying to run your very interesting model on the input images from the project page, but the generated IRs are always kind of the same and do not resemble your output examples.

This is the code I used:

```python
import numpy as np
import torch
from PIL import Image
import torchvision.transforms as transforms

from image2reverb.model import Image2Reverb
from image2reverb.stft import STFT

checkpoint_path = "./checkpoints/model.ckpt"
encoder_path = "./checkpoints/resnet50_places365.pth.tar"
depthmodel_path = "./checkpoints/mono_640x192"
constant_depth = None
latent_dimension = 512

model = Image2Reverb(encoder_path, depthmodel_path, constant_depth=constant_depth, spec="stft")
m = torch.load(checkpoint_path, map_location=model.device)
model.load_state_dict(m["state_dict"])

image_transforms = transforms.Compose([
    transforms.Resize([224, 224], transforms.functional.InterpolationMode.BICUBIC),
    transforms.ToTensor(),
    transforms.Normalize((0.5, 0.5, 0.5), (0.5, 0.5, 0.5))
])

image_path = "examples/input.5416407f.png"

image = Image.open(image_path).convert("RGB")
label = image_transforms(image).unsqueeze(0)

with torch.no_grad():
    f, img = model.enc.forward(label)

    shape = (
        f.shape[0],
        (latent_dimension - f.shape[1]) if f.shape[1] < latent_dimension else f.shape[1],
        f.shape[2],
        f.shape[3]
    )
    z = torch.cat((f, torch.randn(shape, device=model.device)), 1)

    spec = model.g(z)

stft = STFT()
ir = stft.inverse(spec.squeeze())

import soundfile as sf
sf.write("ir.wav", ir, 22050)
```

It's possible I'm doing something wrong or am missing a step. Any ideas? The generated IR always seems to be the same kind of exponentially decaying noise burst, regardless of what the input image looks like.

**Comment by nikhilsinghmus** [comment]
Hi @hollance, thank you for writing! Let me look into it and try your code, though from a quick glance it looks generally right to me. In the meantime, could you try the [replicate demo](https://replicate.com/nikhilsinghmus/image2reverb) with your images and see if that works more as expected? Note there is stochasticity so the match to the web examples may not be exact. In general we have noticed the performance can be sensitive to small changes.

**Comment by hollance** [comment]
Hi @nikhilsinghmus, thanks for looking into this! 

Unfortunately the Replicate website says, "Uh oh! This model can't be run on Replicate because it was built with a version of Cog that is no longer supported."

I was actually going to build a Hugging Face Spaces demo of your model, which would be similar to the Replicate demo (which I wasn't aware existed).

**Comment by nikhilsinghmus** [comment]
Thank you @hollance! It would be awesome to have a demo on Spaces! Apologies for the replicate demo being outdated, and thanks for letting me know; I'll work on getting that updated and back up too. For the moment, it looks like [this](https://replicate.com/p/pgfgw3tbsjdidiolxxxxndp6gy) example page still works, in case you might be interested to just take a look at it.

**Comment by hollance** [comment]
Cheers, I'll have a look. :-) BTW, to test the model with my script I literally used the images from your project page at https://web.media.mit.edu/~nsingh1/image2reverb/

**Comment by nikhilsinghmus** [comment]
Hi @hollance, sorry about the delay and thanks for your patience! [Here](https://web.media.mit.edu/~nsingh1/files/image2reverb_code.zip) is some code that should work better. This also has updated dependencies, specified in the included `environment.yml`. Steps to reproduce should be:
```bash
wget https://web.media.mit.edu/~nsingh1/files/image2reverb_code.zip
unzip image2reverb_code.zip && cd image2reverb_code
conda env create --file environment.yml && eval "$(conda shell.bash hook)" && conda activate image2reverb_demo

python3 demo_cpu_by_sample.py --images images/*.jpg --test_dir images/ # Individual images
# OR
python3 demo_cpu.py --dataset images # Whole directory
```

Corresponding input/output examples are also [here](https://web.media.mit.edu/~nsingh1/files/image2reverb_output12062022/). I ran a few quick code ablations, and it seems for some reason including the `Trainer` is key to reproducing. I'll investigate a little further why this might be the case, so that it isn't needed every time downstream. Thanks again for your interest!

**Comment by hollance** [comment]
Hi @nikhilsinghmus, that's awesome. Thanks so much for looking into this! 

Do you think that maybe Trainer sets the random seed to a specific value, and that the model depends on this seed (since it adds noise)?

**Comment by nikhilsinghmus** [comment]
Thank you for raising it, @hollance! I don't think so, because (1) re-running the demo script gives distinct but similar results, (2) I believe the convention in PyTorch Lightning is to use `seed_everything()` and set `deterministic=True` in `Trainer` for deterministic operation, and (3) we switched to PL at some point but results were similar before (btw the checkpoint in this version is newly trained with updated dependencies, without setting a seed there either). My best guess is that since the model was trained with PL, there could be some preprocessing that is necessary for the whole pipeline to be restored correctly, but not sure what it could be; hoping to dig into their codebase to try to figure it out.

**Comment by hollance** [comment]
Made a demo for this on Hugging Face Spaces, check it out! 😄  https://huggingface.co/spaces/Matthijs/image2reverb

**Comment by nikhilsinghmus** [comment]
Looks great, thanks so much for setting it up @hollance! Will add it to the README/page :)
